Arrangement_on_surface_2/include/CGAL/Arr_dcel_base.h

```diff
@@ -762,10 +765,18 @@ class Arr_inner_ccb : public In_place_list_base<Arr_inner_ccb<V,H,F> >
   typedef Arr_face<V,H,F>                    Face;
   typedef typename Face::Inner_ccb_iterator  Inner_ccb_iterator;
 
+#ifndef CGAL_ARRANGEMENT_DISABLE_UNION_SET_CCB
+  typedef CGAL::Union_find<Self*>            Uf;
+  typedef typename Uf::handle                Uf_handle;
+#endif
+
 private:
   Face* p_f;                  // The face the contains the CCB in its interior.
   Inner_ccb_iterator  iter;   // The inner CCB identifier.
   bool iter_is_not_singular;
+#ifndef CGAL_ARRANGEMENT_DISABLE_UNION_SET_CCB
+  Uf_handle ufh;
```

Comment:
I was trying to think about a way to avoid storing this extra handles but then I realized that it is only at the CCB level so it is probably useless trying to optimize that. If we want to optimise it, one possible way is to replace `p_f` by `std::union F_or_id { Face* p_f; std::size_t id };` and use the id to also store the face pointer in a vector while sweeping.